### ISSUE TITLE
Hang on GATHERING FACTS (setup module)

### ISSUE THREAD

**Issue by dcbishop** [ansible/ansible#10125]
Happening with both Ansible 1.9 and ansible 1.7.2. Host is Ubuntu 14.10.

Here is a trace running the module manually.

```
LANG=C LC_CTYPE=C /usr/bin/python -m trace --trace /home/hegemon/.ansible/tmp/ansible-tmp-1422521131.23-61156481541441/setup
```

```
 --- modulename: subprocess, funcname: _eintr_retry_call
subprocess.py(474):     while True:
subprocess.py(475):         try:
subprocess.py(476):             return func(*args)
subprocess.py(1318):                 if p2cread is not None and p2cwrite is not None:
subprocess.py(1320):                 if c2pwrite is not None and c2pread is not None:
subprocess.py(1321):                     _close_in_parent(c2pwrite)
 --- modulename: subprocess, funcname: _close_in_parent
subprocess.py(1217):                 os.close(fd)
subprocess.py(1218):                 to_close.remove(fd)
subprocess.py(1322):                 if errwrite is not None and errread is not None:
subprocess.py(1323):                     _close_in_parent(errwrite)
 --- modulename: subprocess, funcname: _close_in_parent
subprocess.py(1217):                 os.close(fd)
subprocess.py(1218):                 to_close.remove(fd)
subprocess.py(1326):                 os.close(errpipe_read)
subprocess.py(1328):             if data != "":
subprocess.py(726):         if mswindows:
subprocess.py(734):         if p2cwrite is not None:
subprocess.py(736):         if c2pread is not None:
subprocess.py(737):             if universal_newlines:
subprocess.py(740):                 self.stdout = os.fdopen(c2pread, 'rb', bufsize)
subprocess.py(741):         if errread is not None:
subprocess.py(742):             if universal_newlines:
subprocess.py(745):                 self.stderr = os.fdopen(errread, 'rb', bufsize)
setup(1411):             stdout = ''
setup(1412):             stderr = ''
setup(1413):             rpipes = [cmd.stdout, cmd.stderr]
setup(1415):             if data:
setup(1421):             while True:
setup(1422):                 rfd, wfd, efd = select.select(rpipes, [], rpipes, 1)
setup(1423):                 if cmd.stdout in rfd:
setup(1428):                 if cmd.stderr in rfd:
setup(1436):                 if (not rpipes or not rfd) and cmd.poll() is not None:
 --- modulename: subprocess, funcname: poll
subprocess.py(803):         return self._internal_poll()
 --- modulename: subprocess, funcname: _internal_poll
subprocess.py(1361):             if self.returncode is None:
subprocess.py(1362):                 try:
subprocess.py(1363):                     pid, sts = _waitpid(self.pid, _WNOHANG)
subprocess.py(1364):                     if pid == self.pid:
subprocess.py(1376):             return self.returncode
setup(1441):                 elif not rpipes and cmd.poll() == None:
setup(1421):             while True:
setup(1422):                 rfd, wfd, efd = select.select(rpipes, [], rpipes, 1)
setup(1423):                 if cmd.stdout in rfd:
setup(1428):                 if cmd.stderr in rfd:
setup(1436):                 if (not rpipes or not rfd) and cmd.poll() is not None:
 --- modulename: subprocess, funcname: poll
subprocess.py(803):         return self._internal_poll()
 --- modulename: subprocess, funcname: _internal_poll
subprocess.py(1361):             if self.returncode is None:
subprocess.py(1362):                 try:
subprocess.py(1363):                     pid, sts = _waitpid(self.pid, _WNOHANG)
subprocess.py(1364):                     if pid == self.pid:
subprocess.py(1376):             return self.returncode
setup(1441):                 elif not rpipes and cmd.poll() == None:
setup(1421):             while True:
setup(1422):                 rfd, wfd, efd = select.select(rpipes, [], rpipes, 1)
setup(1423):                 if cmd.stdout in rfd:
setup(1428):                 if cmd.stderr in rfd:
setup(1436):                 if (not rpipes or not rfd) and cmd.poll() is not None:
 --- modulename: subprocess, funcname: poll
```

Let me know if you need the full trace.

**Comment by bcoca** [comment]
is there anything special on this host? I've run against several ubuntu 14.10 and I cannot reproduce the issue.

**Comment by dcbishop** [comment]
Nothing in particular. It did run the first few times without issues. I have tried rebooting but still have the problem. Only thing that springs to mind is changing grub to boot to text mode but I don't see that being an issue.

**Comment by dcbishop** [comment]
Here's the full trace: https://gist.github.com/anonymous/bcef5f59cdb19b4d4576

**Comment by dcbishop** [comment]
It's fixed itself here, similar to #10463. Don't know if it was a systemd-logind update or not...
